CaesarCipher/CaesarCipher/main.m

```diff
@@ -50,11 +50,49 @@ - (NSString *)decode:(NSString *)string offset:(int)offset {
     return [self encode:string offset: (26 - offset)];
 }
 
+- (BOOL)codeBreaker:(NSString *)string1 withStrings:(NSString *)string2 {
+    
+    BOOL result = NO;
+    
+    for (int i = 1; i <= 25; i++) {
+        
+        NSString * decodedString = [self decode:string1 offset:i];
+        
+        for (int j = 1; j <= 25; j++) {
+            
+            NSString * decodedString2 = [self decode:string2 offset:j];
+            
+            if ([decodedString isEqualToString:decodedString2]) {
+                
+                result = YES;
+                
+                NSLog(@"%@ decoded is %@ using offset %d", string1, decodedString, i);
+                
+                NSLog(@"%@ decoded is %@ using offset %d", string2, decodedString2, j);
+               
+                return result;
+            }
+        }
+    }
+    
+    return result;
+}
+
```

Comment:
One suggestion. This method would read a bit better without the result variable: 

```objective-c
- (BOOL)codeBreaker:(NSString *)string1 withStrings:(NSString *)string2 {

    for (int i = 1; i <= 25; i++) {        
        NSString * decodedString = [self decode:string1 offset:i];
        
        for (int j = 1; j <= 25; j++) {
            NSString * decodedString2 = [self decode:string2 offset:j];
            
            if ([decodedString isEqualToString:decodedString2]) {
                NSLog(@"%@ decoded is %@ using offset %d", string1, decodedString, i);
                NSLog(@"%@ decoded is %@ using offset %d", string2, decodedString2, j);
               
                return YES;
            }
        }
    }
    
    return NO;
}

```